File: c/algorithm/greedy/minadd/minadd.c
// 921. 使括号有效的最少添加 https://leetcode.cn/problems/minimum-add-to-make-parentheses-valid/
#include <string.h>

int minAddToMakeValid(char *s) {
    int cnt = 0, ans = 0;
    for (int i = 0; i < strlen(s); i++) {
        if (s[i] == '(') {
            cnt++;
        } else {
            if (cnt > 0) {
                cnt--;
            } else {
                ans++;
            }
        }
    }
    return ans + cnt;
}

File: golang/algorithm/greedy/minadd/minadd.go
// 921. 使括号有效的最少添加 https://leetcode.cn/problems/minimum-add-to-make-parentheses-valid/
package main

import "fmt"

func minAddToMakeValid(s string) (ans int) {
	cnt := 0
	for _, ch := range s {
		if ch == '(' {
			cnt++
		} else {
			if cnt > 0 {
				cnt--
			} else {
				ans++
			}
		}
	}
	return ans + cnt
}

func main() {
	fmt.Println(minAddToMakeValid("))("))
}
